src/openttd.cpp

```diff
@@ -708,14 +708,13 @@ int openttd_main(int argc, char *argv[])
 
 	if (resolution.width != 0) _cur_resolution = resolution;
 
-	/*
-	 * The width and height must be at least 1 pixel and width times
-	 * height times bytes per pixel must still fit within a 32 bits
-	 * integer, even for 32 bpp video modes. This way all internal
-	 * drawing routines work correctly.
-	 */
-	_cur_resolution.width  = ClampU(_cur_resolution.width,  1, UINT16_MAX / 2);
-	_cur_resolution.height = ClampU(_cur_resolution.height, 1, UINT16_MAX / 2);
+	/* Limit resolution as width times height times bytes per pixel must still
+	 * fit within a 32 bits integer, even for 32 bpp video modes. This way all
+	 * internal drawing routines work correctly.
```

Comment:
Difficult to read sentence. Suggestion:

`Limit the width x height x bytes per pixel to fit in a 32bit integer. This way all internal drawing routines work correctly.`